subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/fingerprint/ConfigurationCacheFingerprint.kt

```diff
@@ -47,7 +47,12 @@ sealed class ConfigurationCacheFingerprint {
          * The file system paths that will be ignored during file system checks tracking for the cache fingerprint.
          * @see org.gradle.configurationcache.DefaultIgnoredConfigurationInputs
          */
-        val ignoredFileSystemCheckInputPaths: String?
+        val ignoredFileSystemCheckInputPaths: String?,
+        /**
+         * Without taking Isolated Projects feature option into account a configuration cache entry can be re-used
+         * that does not comply with the additional feature requirements for the build.
+         */
+        val isIsolatedProjectsEnabled: Boolean,
```

Comment:
I have not aware of that option. Could you provide a pointer to where we do this for other values?

What would be the holistic and technical differences between these two alternatives?

This approach seemed like a good fit, because, for instance, we include start parameters in the `GradleEnvironment` fingerprint.

Reply:
See how we do it for `startParameter.isOffline`, for example, https://github.com/gradle/gradle/blob/baf50d7ad43d3bad6a00a9688c954a0e0baeae6e/subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/ConfigurationCacheKey.kt#L75

As for the difference, it's mainly a trade-off between increased likelihood of cache hits and storage requirements. Making the option part of the cache key increases the likelihood of a cache hit (as there are separate cache entries for each possible value) but requires more storage space (and make cache cleanup more expensive).